core/src/test/java/org/elasticsearch/action/index/IndexTaskStatusIT.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.action.index;
+
+import org.elasticsearch.action.ListenableActionFuture;
+import org.elasticsearch.action.admin.cluster.node.tasks.list.ListTasksResponse;
+import org.elasticsearch.action.admin.cluster.node.tasks.list.TaskInfo;
+import org.elasticsearch.test.ESIntegTestCase;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Very basic "is it plugged in" style test that indexes documents and makes
+ * sure that you can fetch the status of the process. The goal here is to verify
+ * that the large moving parts that make fetching task status work fit together
+ * rather than to verify any particular status results from indexing. For that,
+ * look at
+ * {@link org.elasticsearch.action.support.replication.TransportReplicationActionTests}
+ * .
+ */
+public class IndexTaskStatusIT extends ESIntegTestCase {
+    public void testCanFetchIndexStatus() throws InterruptedException, ExecutionException, IOException {
+        long end = System.nanoTime() + TimeUnit.SECONDS.toNanos(30);
+        /*
+         * We try to see an index status. They are pretty quick but we spawn
+         * them continually until we catch one.
+         */
+        ListTasksResponse tasks = null;
+        List<ListenableActionFuture<?>> futuresToFinish = new ArrayList<>();
+        while (true) {
+            futuresToFinish.add(client().prepareIndex("test", "test").setSource("test", "test").execute());
+            tasks = client().admin().cluster().prepareListTasks().setActions("indices:data/write/index*").setDetailed(true).get();
```

Comment:
I'm not super comfortable with this way to check this. I'd prefer some way that forces it to happen the first time but I couldn't figure out one.